From 22ba1c777ebc794859dfeaf6b739ed9efc911181 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 18 Oct 2018 17:48:44 +0300
Subject: [PATCH 0716/1051] ARM64: dts: armada-8040-db-G: Add dual PCIe x4
 board setup G

Add DTS file for Armada-8040-DB setup G.
This configuration includes 1 PCIe x4 and one PCIe x1 port
on each of CP0 and CP1.

Change-Id: Ib402d43fe90057c01b0b28efc173e08e430ac6ec
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60384
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile          |   1 +
 .../boot/dts/marvell/armada-8040-db-G.dts     | 145 ++++++++++++++++++
 .../boot/dts/marvell/armada-8040-db.dtsi      |  29 +++-
 3 files changed, 170 insertions(+), 5 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-G.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 4f0d459aa21d..3a96b5c1ca34 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -14,6 +14,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-G.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8080-pd.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-G.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-G.dts
new file mode 100644
index 000000000000..a3922a3ae2cf
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-G.dts
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This file supports option B configuration
+ * AP_UART,AP_eMMC,CP0_I2C,CP0_eMMC,CP0_RGMII1,CP1_SPI(Boot),CP1_RGMII0
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada 8040 development board G setup";
+	compatible = "marvell,armada8040-db-B", "marvell,armada8040-db",
+		"marvell,armada8040", "marvell,armada-ap806-quad",
+		"marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth2;
+		ethernet1 = &cp1_eth1;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	status = "okay";
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+};
+
+/* CON6 on CP0 expansion */
+&cp0_pcie0 {
+	num-lanes = <4>;
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
+};
+
+/* CON5 on CP0 expansion */
+&cp0_pcie2 {
+	status = "okay";
+	phys = <&cp0_comphy5 2>;
+};
+
+/* CON9 on CP0 expansion */
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+/* CON10 on CP0 expansion */
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp1_spi1 {
+	status = "okay";
+};
+
+/* CON6 on CP1 expansion */
+&cp1_pcie0 {
+	num-lanes = <4>;
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy0 0
+		&cp1_comphy1 0
+		&cp1_comphy2 0
+		&cp1_comphy3 0>;
+};
+
+/* CON5 on CP1 expansion */
+&cp1_pcie2 {
+	status = "okay";
+	phys = <&cp1_comphy5 2>;
+};
+
+/* CON9 on CP1 expansion */
+&cp1_usb3_0 {
+	status = "okay";
+};
+
+/* CON10 on CP0 expansion */
+&cp1_usb3_1 {
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy4 1>;
+	phy-names = "usb";
+	status = "okay";
+};
+
+&cp1_mdio {
+	status = "okay";
+
+	cp1_phy1: ethernet-phy@2 {
+		reg = <0>;
+	};
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+&cp1_eth1 {
+	status = "okay";
+	phy = <&cp1_phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp0_crypto {
+	status = "okay";
+};
+
+&cp1_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index f9fb1ab96511..5673d974c17b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -53,6 +53,25 @@
 		gpio = <&expander1 0 GPIO_ACTIVE_HIGH>;
 	};
 
+	cp1_reg_usb3_1_vbus: cp1-usb3-1-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "cp1-usb3h1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&expander1 1 GPIO_ACTIVE_HIGH>;
+	};
+
+	cp1_usb3_0_phy: cp1-usb3-0-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&cp1_reg_usb3_0_vbus>;
+	};
+
+	cp1_usb3_1_phy: cp1-usb3-1-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&cp1_reg_usb3_1_vbus>;
+	};
+
 	cp0_vccq_sd0_reg: cp0_vccq_sd0 {
 		compatible = "regulator-gpio";
 		regulator-name = "cp0-vccq-sd0";
@@ -66,11 +85,6 @@
 		enable-active-high;
 	};
 
-	cp1_usb3_0_phy: cp1-usb3-0-phy {
-		compatible = "usb-nop-xceiv";
-		vcc-supply = <&cp1_reg_usb3_0_vbus>;
-	};
-
 	sfp_eth0: sfp-eth0 {
 		compatible = "sff,sfp";
 	};
@@ -216,3 +230,8 @@
 &cp1_usb3_0 {
 	usb-phy = <&cp1_usb3_0_phy>;
 };
+
+/* CON10 on CP0 expansion */
+&cp1_usb3_1 {
+	usb-phy = <&cp1_usb3_1_phy>;
+};
-- 
2.17.1

